Scripts/RemediationScripts/Remediate-EnableTransparentDataEncryption.ps1

```diff
@@ -0,0 +1,653 @@
+        [Parameter(HelpMessage="Specifies a forceful roll back without any prompts")]
+        $Force,
+
+        [Switch]
+        [Parameter(HelpMessage="Specifies validation of prerequisites for the command")]
+        $PerformPreReqCheck,
+
+        [String]
+        [Parameter(Mandatory = $true, HelpMessage="Specifies the path to the file to be used as input for the roll back")]
+        $FilePath
+    )
+
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 1 of 3] Preparing to disable Transparent Data Encryption (TDE) for SQL Servers in Subscription: $($SubscriptionId)"
+
+    if ($PerformPreReqCheck)
+    {
+        try
+        {
+            Write-Host "Setting up prerequisites..."
+            Setup-Prerequisites
+        }
+        catch
+        {
+            Write-Host "Error occurred while setting up prerequisites. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+            break
+        }
+    }
+
+    # Connect to Azure account
+    $context = Get-AzContext
+
+    if ([String]::IsNullOrWhiteSpace($context))
+    {
+        Write-Host "Connecting to Azure account..."
+        Connect-AzAccount -Subscription $SubscriptionId -ErrorAction Stop | Out-Null
+        Write-Host "Connected to Azure account." -ForegroundColor $([Constants]::MessageType.Update)
+    }
+
+    # Setting up context for the current Subscription.
+    $context = Set-AzContext -SubscriptionId $SubscriptionId -ErrorAction Stop
+    
+    Write-Host $([Constants]::SingleDashLine)
+    Write-Host "Subscription Name: $($context.Subscription.Name)"
+    Write-Host "Subscription ID: $($context.Subscription.SubscriptionId)"
+    Write-Host "Account Name: $($context.Account.Id)"
+    Write-Host "Account Type: $($context.Account.Type)"
+    Write-Host $([Constants]::SingleDashLine)
+
+    Write-Host "Checking if $($context.Account.Id) is allowed to run this script..."
+
+    # Checking if the current account type is "User"
+    if ($context.Account.Type -ne "User")
+    {
+        Write-Host "WARNING: This script can only be run by `User` Account Type. Account Type of $($context.Account.Id) is: $($context.Account.Type)" -ForegroundColor $([Constants]::MessageType.Warning)
+        break
+    }
+
+    Write-Host "*** To disable Transparent Data Encryption (TDE) for SQL Servers in a Subscription, Contributor and higher privileges on the SQL Servers are required. ***" -ForegroundColor $([Constants]::MessageType.Info)
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 2 of 3] Preparing to fetch all SQL Server database details..."
+    
+    if (-not (Test-Path -Path $FilePath))
+    {
+        Write-Host "ERROR: Input file - $($FilePath) not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+        break
+    }
+
+    Write-Host "Fetching all SQL Server database details from $($FilePath)" -ForegroundColor $([Constants]::MessageType.Info)
+
+    $sqlServerDetails = Import-Csv -LiteralPath $FilePath
```

Comment:
`SQLServers` and `SQLDatabases` are used interchangeably. Uniformity across variables/filenames, etc., will help.

Reply:
Done